src/core/unit.c

```diff
@@ -2138,10 +2148,18 @@ void unit_trigger_notify(Unit *u) {
                         UNIT_VTABLE(other)->trigger_notify(other, u);
 }
 
+static void raise_level(int *log_level, bool condition_info, bool condition_notice) {
```

Comment:
I'd always avoid parameters that are both input and output, if it's easy. Maybe change signature so that the new value is returned and you instead:

```c
level = raise_level(level, …, …);
```

I think that's much nicer, and has this beautiful functional programming touch to it